lib/shotstack/configuration.rb

```diff
@@ -171,8 +187,12 @@ def base_path=(base_path)
       @base_path = '' if @base_path == '/'
     end
 
-    def base_url
-      "#{scheme}://#{[host, base_path].join('/').gsub(/\/+/, '/')}".sub(/\/+\z/, '')
+    # Returns base URL for specified operation based on server settings
+    def base_url(operation = nil)
+      index = server_operation_index.fetch(operation, server_index)
+      return "#{scheme}://#{[host, base_path].join('/').gsub(/\/+/, '/')}".sub(/\/+\z/, '') if index == nil
+
+      server_url(index, server_operation_variables.fetch(operation, server_variables), operation_server_settings[operation])
```

Comment:
I think the problem is in these lines. When the index is not `nil`, this method will call `server_url` and from what I can see the `server_url` doesn't use `base_path` at all.

Reply:
Thanks, I've removed the if statement so that the server_url is never called. I think otherwise this will introduce a breaking change and it isn't needed. Thanks @rudylee.